Doc: Applied admontions to global API (#12776)
facebook/jest: docs/GlobalAPI.md

Runs a function after all the tests in this file have completed. If the function returns a promise or is a generator, Jest waits for that promise to resolve before continuing.

> Optionally, you can provide a `timeout` (in milliseconds) for specifying how long to wait before aborting.
> 
> :::note
> 
> The default timeout is 5 seconds.
> 
> :::

Reviewer: Hm.. Perhaps all timeout notes should stay inline? In this case separate block feels unnecessary, they create to much visual noise.

```suggestion
Optionally, you can provide a `timeout` (in milliseconds) for specifying how long to wait before aborting. The default timeout is 5 seconds.
```

Runs a function after each one of the tests in this file completes. If the function returns a promise or is a generator, Jest waits for that promise to resolve before continuing.

Optionally, you can provide a `timeout` (in milliseconds) for specifying how long to wait before aborting.

:::note

The default timeout is 5 seconds.

> :::

Reviewer: Perhaps all should be reworked for consistency?

Reviewer 2: they should 👍

Reviewer 2: @Biki-das could you update this?

Author: Yeah sure caught up in something will do it soon

#### 1. `describe.each(table)(name, fn, timeout)`

- `table`: `Array` of Arrays with the arguments that are passed into the `fn` for each row. If you pass in a 1D array of primitives, internally it will be mapped to a table i.e. `[1, 2, 3] -> [[1], [2], [3]]`.

  :::note

  If you pass in a 1D array of primitives, internally it will be mapped to a table i.e. `[1, 2, 3] -> [[1], [2], [3]]`

>   :::

Reviewer: Duplication?

#### 1. `describe.each(table)(name, fn, timeout)`

> - `table`: `Array` of Arrays with the arguments that are passed into the `fn` for each row.

Reviewer: Also here it could stay inline:

```suggestion
- `table`: `Array` of Arrays with the arguments that are passed into the `fn` for each row. If you pass in a 1D array of primitives, internally it will be mapped to a table i.e. `[1, 2, 3] -> [[1], [2], [3]]`.
```

Use `test.concurrent` if you want the test to run concurrently.

> :::note
> 
> `test.concurrent` is considered experimental - see [here](https://github.com/facebook/jest/labels/Area%3A%20Concurrent) for details on missing features and other issues
> 
> :::

Reviewer: ```suggestion
::: caution

`test.concurrent` is considered experimental - see [here](https://github.com/facebook/jest/labels/Area%3A%20Concurrent) for details on missing features and other issues.

:::
```

});
```

> :::note
> 
> Use `maxConcurrency` in configuration to prevents Jest from executing more than the specified amount of tests at the same time
> 
> :::

Reviewer: ```suggestion
:::tip

You can use [`maxConcurrency`](Configuration.md#maxconcurrency-number) configuration option to limiting the number of tests that are allowed to run at the same time.

:::
```

#### 1. `test.concurrent.each(table)(name, fn, timeout)`

- `table`: `Array` of Arrays with the arguments that are passed into the test `fn` for each row. If you pass in a 1D array of primitives, internally it will be mapped to a table i.e. `[1, 2, 3] -> [[1], [2], [3]]`.

  :::note

  If you pass in a 1D array of primitives, internally it will be mapped to a table i.e. `[1, 2, 3] -> [[1], [2], [3]]`

  :::

> 

Reviewer: Should this be consistent in all cases?

#### 1. `test.concurrent.each(table)(name, fn, timeout)`

> - `table`: `Array` of Arrays with the arguments that are passed into the test `fn` for each row.

Reviewer: I would keep this inline as well:

```suggestion
- `table`: `Array` of Arrays with the arguments that are passed into the test `fn` for each row. If you pass in a 1D array of primitives, internally it will be mapped to a table i.e. `[1, 2, 3] -> [[1], [2], [3]]`.
```

Use `test.todo` when you are planning on writing tests. These tests will be highlighted in the summary output at the end so you know how many tests you still need todo.

> :::note

Reviewer: ```suggestion
:::tip
```
